src/content/reducers/profile-view.js

```diff
@@ -7,39 +7,45 @@ import * as URLState from './url-state';
 import * as ProfileData from '../profile-data';
 import * as StackTiming from '../stack-timing';
 import * as ProfileTree from '../profile-tree';
-import * as TaskTracer from '../task-tracer';
+import * as TaskTracerTools from '../task-tracer';
 import { getCategoryColorStrategy } from './flame-chart';
 
 import type {
   Profile,
   Thread,
   ThreadIndex,
   IndexIntoFuncTable,
-  IndexIntoMarkersTable,
   SamplesTable,
+  TaskTracer,
 } from '../../common/types/profile';
 import type { TracingMarker, FuncStackInfo, IndexIntoFuncStackTable } from '../../common/types/profile-derived';
 import type { Milliseconds, StartEndRange } from '../../common/types/units';
 import type { Action, CallTreeFilter, ProfileSelection } from '../actions/types';
-import type { State, Reducer, ProfileViewState } from './types';
+import type {
+  State,
+  Reducer,
+  ProfileViewState,
+  RequestedLib,
+  SymbolicationStatus,
+  ThreadViewOptions,
+} from './types';
 
-type RequestedLib = { pdbName: string, breakpadId: string };
-type SymbolicationStatus = 'DONE' | 'SYMBOLICATING';
-type ThreadViewOptions = {
-  selectedFuncStack: IndexIntoFuncTable[],
-  expandedFuncStacks: Array<IndexIntoFuncTable[]>,
-  selectedMarker: IndexIntoMarkersTable | -1,
+function emptyProfile(): Profile {
+  return {
+    meta: { interval: 1 },
+    threads: [],
+    tasktracer: TaskTracerTools.emptyTaskTracerData(),
+  };
 }
-
-function profile(state: Profile | null = null, action: Action) {
+function profile(state: Profile = emptyProfile(), action: Action) {
```

Comment:
Throwing this out there, but I wonder if you could have two different base state types, an `InitialState` type without a profile, and then a `State` with a profile.